src/textures/CanvasTexture.js

```diff
@@ -1,15 +1,17 @@
 import { Texture } from './Texture.js';
 
-function CanvasTexture( canvas, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy ) {
+class CanvasTexture extends Texture {
 
-	Texture.call( this, canvas, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy );
+	constructor( canvas, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy ) {
 
-	this.needsUpdate = true;
+		super( canvas, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy );
 
-}
+		Object.defineProperty( this, 'isCanvasTexture', { value: true } );
```

Comment:
Use this pattern:
```js
CanvasTexture.prototype. isCanvasTexture = true;
```
for all texture classes.

Reply:
roger